### ISSUE TITLE
C0103 false positive when importing tomli_w

### ISSUE THREAD

**Issue by deadc0de6** [pylint-dev/pylint#8930]
### Bug description

Running pylint in the following files reports a C0103:
```python
'''
cool
'''
import tomli_w

if __name__ == '__main__':
    out = tomli_w.dumps("")
    print(out)
```

However, when using `toml` instead with the same kind of construct, no C0103 is reported
```python
'''
cool
'''
import toml

if __name__ == '__main__':
    out = toml.dumps("test")
    print(out)
```

I'm however not really sure why this happens?

### Command used

```shell
pylint file.py
```

### Pylint output

```shell
************* Module file
file.py:7:4: C0103: Constant name "out" doesn't conform to UPPER_CASE naming style (invalid-name)

------------------------------------------------------------------
Your code has been rated at 7.50/10 (previous run: 7.50/10, +0.00)
```

### Expected behavior

--------------------------------------------------------------------
Your code has been rated at 10.00/10 (previous run: 10.00/10, +0.00)

### Pylint version

```shell
pylint 2.17.5
astroid 2.15.6
Python 3.11.4
```

**Comment by parona-source** [comment]
To avoid possible confusion, ill remove some leftovers I left in the test example.

```python
'''
with toml

--------------------------------------------------------------------
Your code has been rated at 10.00/10 (previous run: 0.00/10, +10.00)
'''
import toml

if __name__ == '__main__':
    out = toml.dumps("")
    print(out)
```

```python
'''
with tomli_w

************* Module t
t.py:7:4: C0103: Constant name "out" doesn't conform to UPPER_CASE naming style (invalid-name)

------------------------------------------------------------------
Your code has been rated at 7.50/10 (previous run: 7.50/10, +0.00)
'''
import tomli_w

if __name__ == '__main__':
    out = tomli_w.dumps("")
    print(out)
```

**Comment by mbyrnepr2** [comment]
Thanks for the issue folks. @deadc0de6 @parona-source I'm inclined to close this as "not planned" since the issue here is that Pylint can't infer the [toml](https://pypi.org/project/toml/) package's `dump` object. The typical way of dealing with these issues is that we would either add an astroid brain for that package (Pylint's inference lib) or ideally, create a plugin for this which can be independently maintained. However, since there is no recent active development for `toml` and since there is `tomllib` in Python 3.11, I suggest using that instead - what do you think?

The other point is that this is actually a false negative for the toml package, since Pylint expects `out` to be `OUT` in this context (treated like a constant).

I'll leave the issue open in case there are other views on what I've said.

```python
import tomllib

if __name__ == '__main__':
    toml_string = """
python-version = "3.11.0"
"""
    out = tomllib.loads(toml_string)
    print(type(out))
```

**Comment by parona-source** [comment]
tomllib isn't able to write TOML files so its not an option there. 

But if this was false negative from toml rather than a false positive from tomli_w then there is no problem in actuality for our case. As the change that triggered this was a move from toml to tomllib/tomli+tomli_w.

**Comment by deadc0de6** [comment]
@mbyrnepr2 thanks for your answer but I think we should keep `toml` out of the loop. The following example is indeed producing a false positive when using `pylint`:

```python
"""
docstring
"""
import tomli_w

if __name__ == '__main__':
    doc = {"abc": 1}
    out = tomli_w.dumps(doc)
    print(out)
```

```bash
$ python3 /tmp/xyz.py
abc = 1
$ pylint /tmp/xyz.py
************* Module xyz
/tmp/xyz.py:8:4: C0103: Constant name "out" doesn't conform to UPPER_CASE naming style (invalid-name)

------------------------------------------------------------------
Your code has been rated at 8.00/10 (previous run: 6.00/10, +2.00)
```

At least I don't expect `pylint` to consider `out` as a global constant inside `if __name__ == '__main__':` or am I missing something?

**Comment by mbyrnepr2** [comment]
> @mbyrnepr2 thanks for your answer but I think we should keep `toml` out of the loop. The following example is indeed producing a false positive when using `pylint`:
> 
> ```python
> """
> docstring
> """
> import tomli_w
> 
> if __name__ == '__main__':
>     doc = {"abc": 1}
>     out = tomli_w.dumps(doc)
>     print(out)
> ```
> 
>  ```shell
> $ python3 /tmp/xyz.py
> abc = 1
> $ pylint /tmp/xyz.py
> ************* Module xyz
> /tmp/xyz.py:8:4: C0103: Constant name "out" doesn't conform to UPPER_CASE naming style (invalid-name)
> 
> ------------------------------------------------------------------
> Your code has been rated at 8.00/10 (previous run: 6.00/10, +2.00)
> ```
> 
> At least I don't expect `pylint` to consider `out` as a global constant inside `if __name__ == '__main__':` or am I missing something?

It's just a module-level constant, right? https://pylint.readthedocs.io/en/stable/user_guide/messages/convention/invalid-name.html. I'm unsure why it isn't matching your expectations (not saying you don't have a point :D).

**Comment by deadc0de6** [comment]
@mbyrnepr2 indeed you're right, point taken ;-)
Thanks again for your time on this.
